<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2026-02-17">

<title>Parametrizations – trainingMM</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../about.html" rel="next">
<link href="../pages/03.1_predict_LMM.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>

<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/08_Parametrizations.html">Parametrizations</a></li><li class="breadcrumb-item"><a href="../pages/08_Parametrizations.html">Breeding vs Genotypic</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">trainingMM</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/AparicioJohan/trainingMM" title="GitHub repository" class="quarto-navigation-tool px-1" aria-label="GitHub repository"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Starting LM</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/00_start_lm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ANOVA and fixed effects</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/01_predict_LM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Marginal means (SE)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/02_QR_decomp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">QR decomposition in LM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/04_covariates_LM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Covariates in LM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/99_404.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Other software</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Starting LMM</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/05_MVN.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multivariate Normal</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/03_LMM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mixed model by hand</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/06_Heritabilities.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Heritability</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/03.1_predict_LMM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predictions in LMM</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/99_404.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Outliers LMM</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Parametrizations</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/08_Parametrizations.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Breeding vs Genotypic</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/08_Parametrizations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Polyploids and Digenic Dominance</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Advanced LMM</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/99_404.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genotype-by-environment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/99_404.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Factor analytic models</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/99_404.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Two-stage analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../about.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#diploid-theory" id="toc-diploid-theory" class="nav-link active" data-scroll-target="#diploid-theory">Diploid Theory</a>
  <ul class="collapse">
  <li><a href="#breeding-parametrization" id="toc-breeding-parametrization" class="nav-link" data-scroll-target="#breeding-parametrization">1) Breeding Parametrization</a></li>
  <li><a href="#genotypic-parametrization" id="toc-genotypic-parametrization" class="nav-link" data-scroll-target="#genotypic-parametrization">2) Genotypic Parametrization</a></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">3) Summary</a></li>
  </ul></li>
  <li><a href="#autoteraploid-theory" id="toc-autoteraploid-theory" class="nav-link" data-scroll-target="#autoteraploid-theory">Autoteraploid Theory</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../pages/08_Parametrizations.html">Parametrizations</a></li><li class="breadcrumb-item"><a href="../pages/08_Parametrizations.html">Breeding vs Genotypic</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Parametrizations</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 17, 2026</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<blockquote class="blockquote">
<p>In this chapter we will go through the derivation and assumptions of the two most common parametrizations to model additivity and dominance for both diploids and autotetraploid species. Please, refer to <a href="https://academic.oup.com/genetics/article/195/4/1223/5935377?login=false">Vitezica et al, 2013</a> and <a href="https://doi.org/10.1534/genetics.118.300685">Endelman et al, 2018</a> for further explanations.</p>
</blockquote>
<section id="diploid-theory" class="level1">
<h1>Diploid Theory</h1>
<p>Let’s begin by defining the following model</p>
<p><span class="math display">\[
y = \mu + Ta + Xd + \varepsilon
\]</span></p>
<p>The frequencies and genetic values of each possible genotype for a single locus under Hardy-Weinberg equilibrium can be defined as</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-striped caption-top table-sm small" data-quarto-postprocess="true">
<caption>Genotypic Values For a Loxus</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Genotype</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Frequency</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">$A_1A_1$</td>
<td style="text-align: left;">$q^2$</td>
<td style="text-align: left;">$- a$</td>
</tr>
<tr class="even">
<td style="text-align: left;">$A_1A_2$</td>
<td style="text-align: left;">$2pq$</td>
<td style="text-align: left;">$d$</td>
</tr>
<tr class="odd">
<td style="text-align: left;">$A_2A_2$</td>
<td style="text-align: left;">$p^2$</td>
<td style="text-align: left;">$a$</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Then, the genetic mean of the population under both parametrizations equals</p>
<p><span class="math display">\[
\begin{align*}
\mu_G &amp;= \sum (\text{value} \times \text{frequency}) = \\
     &amp;= P_{A_1, A_1}*G_{A_1, A_1} + P_{A_1, A_2}*G_{A_1, A_2} + P_{A_2, A_2}*G_{A_2, A_2}\\
     &amp;= p^2(-a) + 2pq(d) + q^2(a) \\
     &amp;= a(p^2 - q^2) + 2pqd \\
     &amp;= a((p - q)(p + q)) + 2pqd \\
     &amp;= a(p - q) + 2pqd
\end{align*} \\
\]</span></p>
<section id="breeding-parametrization" class="level2">
<h2 class="anchored" data-anchor-id="breeding-parametrization">1) Breeding Parametrization</h2>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">Theory</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Proofs</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<p>This approach separate the Genotypic Value (<span class="math inline">\(G\)</span>) of an individual into the breeding value (<span class="math inline">\(u\)</span>) and dominance deviations (<span class="math inline">\(v\)</span>)</p>
<p><span class="math display">\[
G = \mu_G + u + v
\]</span></p>
<section id="breeding-values-u" class="level4">
<h4 class="anchored" data-anchor-id="breeding-values-u">Breeding Values (<span class="math inline">\(u\)</span>)</h4>
<p>This value is defined as a function of the additive substitution effect <span class="math inline">\(\alpha\)</span>. This parameter is computed as the difference of the alleles substitution effects, which represent the mean deviation from the population mean of the individuals receiving this particular allele from one parent and the second allele at random from the population.</p>
<p><span class="math display">\[
\begin{align*}
\alpha &amp;=  a + d(q - p)
\end{align*}
\]</span></p>
<p>Thus, for a single locus with two alleles</p>
<p><span class="math display">\[
u =
\begin{cases}
    (0 - 2p)\alpha = -2p\alpha\\
    (1 - 2p)\alpha = (q - p)\alpha\\
    (2 - 2p)\alpha = 2q\alpha\\
\end{cases}
\]</span></p>
<p>So, the breeding values of a set of individuals is <span class="math inline">\(\mathbf{u} = \mathbf{Z}\alpha = \mathbf{Z}a + \mathbf{Z}(q - p)d\)</span> where <span class="math inline">\(\mathbf{Z}\)</span> is the centered marker matrix, so that <span class="math inline">\(\mathbf{Z} = (z_1...z_n)\)</span> including all markers is identical to <span class="math inline">\(\mathbf{T}\)</span> but centered. Then</p>
<p><span class="math display">\[
z_i =
\begin{cases}
    -2p, &amp; \text{for } A_1A_1\\
    q - p, &amp; \text{for } A_1A_2\\
     2q, &amp; \text{for } A_2A_2
\end{cases}
\]</span> #### Dominance Deviations</p>
<p>Alternatively, the dominance deviations can be computed as the difference between the actual genotypic value and the genotypic value predicted from the sum of the average effects</p>
<p><span class="math display">\[
\begin{align*}
G_{ij} &amp;= \mu_G + \alpha_i + \alpha_j + v{ij} \\
\hat{G}_{ij} &amp;= \mu_G + \alpha_i + \alpha_j \\
v_{ij} &amp;= G_{ij} - \hat{G}_{ij} \\
             &amp;= G_{ij} - (\mu_G + \alpha_i + \alpha_j) \\
             &amp;= G_{ij} - \mu_G - u_{ij}
\end{align*}
\]</span></p>
<p>Then</p>
<p><span class="math display">\[
v =
\begin{cases}
    - 2p^2d \\
    2pqd\\
    - 2q^2d
\end{cases}
\]</span></p>
<p>In summary, <span class="math inline">\(\mathbf{v} = \mathbf{W}d\)</span>, where <span class="math inline">\(W = (w_1...w_n)\)</span> is not <span class="math inline">\(\mathbf{X}\)</span> in the model and</p>
<p><span class="math display">\[
w_i =
\begin{cases}
    - 2p^2,  &amp; \text{for } A_1A_1\\\\
    2pq,  &amp; \text{for } A_1A_2\\\\
    -2q^,  &amp; \text{for } A_2A_2\\
\end{cases}
\]</span></p>
<p>Besides that, it is important to note that the expected Value of both <span class="math inline">\(u\)</span> and <span class="math inline">\(v\)</span> equals 0. Under this parametrization, we will assume that there is no covariance between additive and dominance, and therefore we can also express the genetic variance for a single locus simply as</p>
<p><span class="math display">\[
\operatorname{var}(G) = \operatorname{var}(A) + \operatorname{var}(D)  
\]</span></p>
<p>Where</p>
<p><span class="math display">\[
\operatorname{var}(A) = 2pq[a + d(1-p)]^2 + (2pqd)^2
\]</span></p>
<p><span class="math display">\[
\operatorname{var}(D) = (2pqd)^2
\]</span></p>
<p>Besides that, if we assume that <span class="math inline">\(a \sim N(0, \sigma^2_a)\)</span> and <span class="math inline">\(d \sim N(0, \sigma^2_d)\)</span>, and that <span class="math inline">\(\sigma_A^2 = 2pq\big(\sigma_a^2 + (q - p)^2\sigma_d^2\big)\)</span></p>
<p><span class="math display">\[
\begin{align*}
\operatorname{cov}(\mathbf{u})
&amp;=\operatorname{cov}(\mathbf{Z}\alpha) \\
&amp;= \operatorname{var}(\mathbf{Z}(\mathbf{a} + \mathbf{d}(p-q))) \\
&amp;= \mathbf{ZZ'}\operatorname{var}(\mathbf{a}) + \mathbf{ZZ'}(p-q)^2\operatorname{var}(\mathbf{d}) \\
  &amp;=  \mathbf{ZZ'}\sigma^2_a + \mathbf{ZZ'}(p-q)^2\sigma^2_d \\
&amp;= \mathbf{ZZ'} \big(\sigma_a^2 + (p-q)^2 \sigma_d^2\big) \\
&amp;= \mathbf{ZZ'} \frac{\sigma_A^2}{2pq} \\
&amp;= \frac{\mathbf{ZZ'}}{2pq} \, \sigma_A^2 \\
&amp;= \mathbf{G} \, \sigma_A^2
\end{align*}
\]</span></p>
<p>Which equals the VanRaden method to compute the Additive Genomic Relationship Matrix. A similar Derivation can be made for dominance deviations, considering that <span class="math inline">\(\operatorname{\sigma^2_D} = (2pq)^2\sigma^2_d\)</span></p>
<p><span class="math display">\[
\begin{align*}
\operatorname{cov}(\mathbf{v})
&amp;= \operatorname{var}(\mathbf{W}\mathbf{d}) \\
&amp;= \mathbf{WW'} \, \sigma_d^2 \\
&amp;= \frac{\mathbf{WW'}}{(2pq)^2} \, \sigma_D^2 \\
&amp;= \mathbf{D} \, \sigma_D^2
\end{align*}
\]</span></p>
</section>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<ol type="1">
<li>Allele substitution effects</li>
</ol>
<p>The allele substitution are by definition the differrence between the mean of the population carrying the allele and the overall population mean. Thus</p>
<p><span class="math display">\[
\alpha_{A_1} = \mu_{A_1} - \mu
\]</span> <span class="math display">\[
\alpha_{A_2} = \mu_{A_2} - \mu
\]</span></p>
<p>So we need to compute <span class="math inline">\(\mu_{A_1}\)</span> and <span class="math inline">\(\mu_{A_2}\)</span></p>
<p>Let’s consider then the average effect of Allele <span class="math inline">\({A_1}\)</span></p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-striped caption-top table-sm small" data-quarto-postprocess="true">
<caption>Genotypic Values For a Loxus</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Allele.from.other.Parent</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Probability</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Genotype</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">$A_1$</td>
<td style="text-align: left;">$p$</td>
<td style="text-align: left;">$A_1A_1$</td>
<td style="text-align: left;">$a$</td>
</tr>
<tr class="even">
<td style="text-align: left;">$A_2$</td>
<td style="text-align: left;">$q$</td>
<td style="text-align: left;">$A_1A_2$</td>
<td style="text-align: left;">$d$</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>And the average effect of Allele <span class="math inline">\(A_2\)</span></p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-striped caption-top table-sm small" data-quarto-postprocess="true">
<caption>Genotypic Values For a Loxus</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Allele.from.other.Parent</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Probability</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Genotype</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">$A_1$</td>
<td style="text-align: left;">$p$</td>
<td style="text-align: left;">$A_1A_2$</td>
<td style="text-align: left;">$d$</td>
</tr>
<tr class="even">
<td style="text-align: left;">$A_2$</td>
<td style="text-align: left;">$q$</td>
<td style="text-align: left;">$A_2A_2$</td>
<td style="text-align: left;">$-a$</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>We can simply compute the subpopulations means as</p>
<p><span class="math display">\[
\mu_{A_1} = ap + dq
\]</span> <span class="math display">\[
\mu_{A_2} = dp - aq
\]</span></p>
<p>We can solve now the allele substitution effects taking into account that the population mean <span class="math inline">\(\mu = a(p - q) + 2pqd\)</span></p>
<p><span class="math display">\[
\begin{align*}
\alpha_{A_1} &amp;= ap + dq - (a(p - q) + 2pqd) \\
         &amp;= ap + dq - ap + aq - 2pqd \\
         &amp;= q(a - 2pd + d) \\
         &amp;= q(a + d(-2p + 1)) \\
         &amp;= q(a + d(-2p + p + q)) \\
         &amp;= q(a + d(q - p))
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
\alpha_{A_2} &amp;= ap + dq - (a(p - q) + 2pqd) \\
         &amp;= dp - aq - ap + aq - 2pqd \\
         &amp;= p(-a + d(-2q + 1)) \\
         &amp;= -p(a + d(2q - (p + q))) \\
         &amp;= -p(a + d(q - p))
\end{align*}
\]</span></p>
<p>Now, if <span class="math inline">\(\alpha = \alpha_{A_1} - \alpha_{A_2}\)</span>, then</p>
<p><span class="math display">\[
\begin{align*}
\alpha &amp;= q(a + d(q - p)) - (-p(a - d(q + p)))\\
       &amp;= a + d(q - p)
\end{align*}
\]</span></p>
<p>Finally, we can express the allele substitution effects of each allele in terms of <span class="math inline">\(\alpha\)</span></p>
<p><span class="math display">\[
\alpha_{A_1} = q(a + d(q - p)) = q\alpha
\]</span></p>
<p><span class="math display">\[
\alpha_{A_2} = -p(a + d(q - p)) = -p\alpha
\]</span></p>
<ol start="2" type="1">
<li>Dominance Deviations</li>
</ol>
<p>Consider first that <span class="math inline">\(v = G - (\mu + u)\)</span>. Then, following the information shown below</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-striped caption-top table-sm small" data-quarto-postprocess="true">
<caption>Genotypic Values For a Loxus</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Genotypes</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Frequency</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Value</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">u</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">$A_1A_1$</td>
<td style="text-align: left;">q^2</td>
<td style="text-align: left;">- a</td>
<td style="text-align: left;">$-2p\alpha$</td>
</tr>
<tr class="even">
<td style="text-align: left;">$A_1A_2$</td>
<td style="text-align: left;">2pq</td>
<td style="text-align: left;">d</td>
<td style="text-align: left;">$(q - p)\alpha$</td>
</tr>
<tr class="odd">
<td style="text-align: left;">$A_2A_2$</td>
<td style="text-align: left;">p^2</td>
<td style="text-align: left;">a</td>
<td style="text-align: left;">$2q\alpha$</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>We can solve that</p>
<p><span class="math display">\[
\begin{align*}
v_{A_1, A_1}
&amp;= -a - (a(p - q) + 2pqd) - (-2p\alpha) \\
    &amp;= -a -ap + aq - 2pqd + 2p(a + d(q - p )) \\
    &amp;= -a -ap + aq - 2pqd + 2pa + 2pdq - 2p^2d \\
    &amp;= -a + aq + pa - 2p^2d \\
    &amp;= -a + a(p + q) - 2p^2d \\
    &amp;= - 2p^2d \\
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
v_{A_1, A_2}
&amp;= d - \big(a(p-q) + 2pq d\big) - (q-p)\big(a + d(q-p)\big) \\
&amp;= d - a(p-q) - 2pq d - (q-p)a - d(q-p)^2 \\
&amp;= d - 2pq d - d(q-p)^2 \\
&amp;= d(1 - 2pq - (q-p)^2) \\
&amp;= d\big(1 - p^2 - q^2\big) \\
&amp;= d(2pq) \\
&amp;= 2pq\,d
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
v_{A_2, A_2} &amp;= a - \big(a(p-q) + 2pq d\big) -2q\alpha \\
&amp;= a -ap + aq -2pqd -2q(a+ d(q-p))\\
&amp;= a -ap + aq -2pqd -2aq -2q^2d + 2pqd \\
&amp;= a - ap -aq -2q^2d \\
&amp;= -2q^2d
\end{align*}
\]</span></p>
<ol start="3" type="1">
<li>Expectations</li>
</ol>
<p>Let’s prove why the expectations of the breeding values and the dominance deviations equals 0.</p>
<p><span class="math display">\[
\begin{align*}
E(u) &amp;= p^2(2q\alpha) + 2pq(q-p)\alpha + q^2(-2p\alpha) \\
  &amp;= 2p^2q\alpha + 2pq^2\alpha - 2p^2q\alpha -2pq^2\alpha \\
  &amp;= 0
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
E(v) &amp;= p^2(-2q^2d) + 2pq2pqd + q^2(-2p^2d) \\
  &amp;= -2p^2q^2d + 4p^2q^2d -2p^2q^2d\\
  &amp;= 0
\end{align*}
\]</span></p>
<ol start="4" type="1">
<li>Variances</li>
</ol>
<p>Following the properties of the variance, we can define the additive and dominance variance as follows</p>
<p><span class="math display">\[
\begin{align*}
\operatorname{var}(A) &amp;= \operatorname{var}(u) = E[(u−E(u))^2]=E(u^2)− \cancel{[E(u)]^2} \\
  &amp;= p^2(2q\alpha)^2 + 2pq((q-p)\alpha)^2 + q^2(-2p\alpha)^2\\
  &amp;= 4p^2q^2\alpha^2 + 2pq(q-p)^2\alpha^2 +4 p^2q^2\alpha^2 \\
  &amp;= 2pq\alpha^2(2pq + (q-p)^2 + 2pq) \\
  &amp;= 2pq\alpha^2(2pq + p^2 + q^2 -2pq + 2pq) \\
  &amp;= 2pq\alpha^2(p^2 + q^2 + 2pq) \\
  &amp;= 2pq\alpha^2 = 2pq[a + d(1-p)]^2
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
\operatorname{var}(D) &amp;= \operatorname{var}(v) = E[(v−E(v))^2]=E(v^2)−\cancel{[E(v)]^2} \\
  &amp;= p^2(-2q^2d)^2 + 2pq(2pqd)^2 + q^2(-2p^2d) ^2 \\
  &amp;= 4p^2q^4d^2 + 8p^3q^3d^2 + 4p^4q^2d^2 \\
  &amp;= 4p^2q^2d^2(q^2 + 2pq + p^2) \\
  &amp;= 4p^2q^2d^2 = (2pqd)^2
\end{align*}
\]</span></p>
</div>
</div>
</div>
</section>
<section id="genotypic-parametrization" class="level2">
<h2 class="anchored" data-anchor-id="genotypic-parametrization">2) Genotypic Parametrization</h2>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">Theory</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">Proofs</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<section id="pure-additive-values" class="level4">
<h4 class="anchored" data-anchor-id="pure-additive-values">Pure additive values</h4>
<p>A different approach is to define the genotypic value as</p>
<p><span class="math display">\[
G = E(G) + u^* + v^*
\]</span></p>
<p>Where <span class="math inline">\(u^*\)</span> and <span class="math inline">\(v^*\)</span> are no longer the breeding values and dominance deviations but pure additive and pure additive values. Given that we aim to partition the effects in pure additive and pure dominance effects, we will no longer have dominance with <span class="math inline">\(u^*\)</span>. An easy way to summarize this is to remove <span class="math inline">\(d\)</span> from <span class="math inline">\(\alpha\)</span>, so that <span class="math inline">\(\alpha = a\)</span>. then, we can substitute to obtain</p>
<p><span class="math display">\[
u^* =
\begin{cases}
    -2pa\\
    (q-p)a\\
    2qa
\end{cases}
\]</span></p>
<p>See how still, the covariate equals the <span class="math inline">\(Z\)</span> matrix as in the Breeding parametrization. Then, we can express <span class="math inline">\(u^* = \mathbf{Z}a\)</span>.</p>
</section>
<section id="pure-dominance-values" class="level4">
<h4 class="anchored" data-anchor-id="pure-dominance-values">Pure dominance values</h4>
<p>The pure dominance values are derived by substracting the population mean, and the pure additive values to the genotypic values. Therefore</p>
<p><span class="math display">\[
v^* =
\begin{cases}
    -2pqd\\
    d(1 -2pq)\\
    -2pqd
\end{cases}
\]</span></p>
<p>So that <span class="math inline">\(\mathbf{v^*} = \mathbf{H}d\)</span> where</p>
<p><span class="math display">\[
\mathbf{H} =
\begin{cases}
    -2pqd,  &amp; \text{for } A_1A_1\\\\
    (1 -2pq)d,  &amp; \text{for } A_1A_2\\\\
    -2pqd,  &amp; \text{for } A_2A_2\\
\end{cases}
\]</span></p>
<p>It is important to notice that <span class="math inline">\(H \neq W\)</span>. However, it can be proved that the expected values of <span class="math inline">\(u^*\)</span> and <span class="math inline">\(v^*\)</span> is still <span class="math inline">\(0\)</span>. Now, the genetic variance equals</p>
<p><span class="math display">\[
\operatorname{var}(G) = 2pqa^2 + 2pq[ 1 - 2pq ]d^2
\]</span></p>
<p>And the covariances between additive values are computed as</p>
<p><span class="math display">\[
\begin{align*}
\operatorname{cov}(\mathbf{u^*}) &amp;= \mathbf{ZZ'}\sigma^2_{a} \\
  &amp;=  \frac{\mathbf{ZZ'}}{2pq}\sigma^2_{A^*} \\
  &amp;= \mathbf{G}\sigma^2_{A^*}
\end{align*}
\]</span></p>
<p>Notice that the Additive GRM follows the same definition for both parametrizations. Let’s show how this is not the case for the Dominance GRM</p>
<p><span class="math display">\[
\begin{align*}
\operatorname{cov}(\mathbf{v^*}) &amp;= \mathbf{HH'}\sigma^2_{d} \\
  &amp;=  \frac{\mathbf{HH'}}{2pq(1 - 2pq)}\sigma^2_{D^*} \\
  &amp;= \mathbf{D^*}\sigma^2_{D^*}
\end{align*}
\]</span></p>
</section>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<ol type="1">
<li>Pure Dominance Values</li>
</ol>
<p>Under the genotypic parametrization, <span class="math inline">\(v^*\)</span> are computed as <span class="math inline">\(v\)</span> in the breeding parametrization, although now, by definition <span class="math inline">\(v^* = G - (\mu + u^*)\)</span>. Despite the population mean remains the same, now <span class="math inline">\(u^* \neq u\)</span>, so</p>
<p><span class="math display">\[
  \begin{align*}
v^*_{A_1A_1}
&amp;= -a - \big(a(p-q) + 2pq d\big) + 2pa \\
&amp;= -a -ap + aq -2pqd + 2pa \\
&amp;= -a + a(p + q) -2pqd \\
&amp;= -2pqd
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
v^*_{A_1A_2}
&amp;= d - \big(a(p-q) + 2pq d\big) -(q-p)a) \\
&amp;= d -ap + aq -2pqd -aq + ap \\
&amp;= d(1 -2pq)
\end{align*}
\]</span></p>
<p><span class="math display">\[
  \begin{align*}
v^*_{A_2A_2}
&amp;= a - \big(a(p-q) + 2pq d\big) -2qa \\
&amp;= a - ap + aq -2pqd -2qa \\
&amp;= a -ap -2pqd -qa \\
&amp;= a -a(p + 1) -2pqd \\
&amp;= -2pqd
\end{align*}
\]</span></p>
<ol start="2" type="1">
<li>Expectations</li>
</ol>
<p>The expectation of <span class="math inline">\(u^*\)</span> and <span class="math inline">\(v^*\)</span> still equal 0</p>
<p><span class="math display">\[
\begin{align*}
E(u^*) &amp;= p^2(2qa) + 2pq(q-p)a + q(-2pa) \\
&amp;= 2p^2qa + 2pq^2a - 2p^2qa -2pq^2a\\
&amp;= 0\\
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
E(v^*) &amp;= p^2(-2pqd) + 2pq((1-2pq)d) + q(-2pqd) \\
&amp;= -2p^2qd + 2pqd - 4p^2q^2d -2pq^2d\\
&amp;= 2pqd(-p^2 + 1 - 2pq - q^2)\\
&amp;= 2pqd(-1 + 1) = 0
\end{align*}
\]</span></p>
<ol start="3" type="1">
<li>Variances</li>
</ol>
<p>By definition, the variances under the genotypic parametrization can be defined as</p>
<p><span class="math display">\[
  \begin{align*}
Var(A^*) = \sigma^2_{A^*} &amp;= E(u^{*2}) - \cancel{E(u^*)^2} \\
&amp;= p^2(2qa)^2 + 2pq((q-p)a)^2 + q^2(-2pa)^2 \\
&amp;= 4p^2q^2a^2 + 2pq(q-p)^2a^2 + 4p^2q^2a^2 \\
&amp;= 2pqa^2(2pq + (q-p)^2 + 2pq) \\
&amp;= 2pqa^2(q^2 + p^2 +2pq)\\
&amp;= 2pqa^2
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
Var(D^*) = \sigma^2_{D^*} &amp;= E(v^{*2}) - \cancel{E(v^*)^2} \\
&amp;= p^2\big(-2pq\,d)^2 \;+\; 2pq((1-2pq)\,d)^2 \;+\; q^2(-2pq\,d)^2 \\
&amp;= d^2[ p^2( -2pq )^2 \;+\; 2pq(1-2pq)^2 \;+\; q^2( -2pq )^2\\
        &amp;= d^2[ 4p^4q^2 \;+\; 2pq(1 - 4pq + 4p^2q^2) \;+\; 4p^2q^4] \\
        &amp;= d^2[ 4p^2q^2(p^2+q^2) \;+\; 2pq(1 - 4pq + 4p^2q^2)] \\
        &amp;= 2pq[ 2pq(p^2+q^2) \;+\; (1 - 4pq + 4p^2q^2)]d^2\\
        &amp;= 2pq[ 2pq(1-2pq) \;+\; 1 - 4pq + 4p^2q^2 ] d^2\\
        &amp;= 2pq[ (2pq - 4p^2q^2) \;+\; (1 - 4pq + 4p^2q^2)d^2]d^2 \\
        &amp;= 2pq[ 1 - 2pq ]d^2 \\
        \end{align*}
\]</span></p>
</div>
</div>
</div>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">3) Summary</h2>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-3-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-1" role="tab" aria-controls="tabset-3-1" aria-selected="true">Theory</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-2" role="tab" aria-controls="tabset-3-2" aria-selected="false">Proofs</a></li></ul>
<div class="tab-content">
<div id="tabset-3-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-3-1-tab">
<p>Briefly, <span class="math inline">\(\operatorname{cov}(\mathbf{u^*}) = \operatorname{cov}(\mathbf{u})\)</span>, but <span class="math inline">\(\operatorname{cov}(\mathbf{v^*}) \neq \operatorname{cov}(\mathbf{v})\)</span>, given that <span class="math inline">\(\mathbf{D^*} \neq \mathbf{D}\)</span>. Besides, given that <span class="math inline">\(\sigma^2_A + \sigma^2_D = \sigma^2_{A^*} + \sigma^2_{D^*}\)</span>. THus, both approaches explains the data but their interpretation must be different.</p>
<p>Finally, we wanted to highlight that from the practical point of view, t he conversion from the genotypic to the breeding values is easy, given that we can use <span class="math inline">\(a\)</span>, <span class="math inline">\(d\)</span> estimated from the model and the allelic frequencies <span class="math inline">\(p\)</span> and <span class="math inline">\(q\)</span> from the original marker matrix to build <span class="math inline">\(\alpha = a + d(q - p)\)</span>. Then we should simply multiply <span class="math inline">\(Z\alpha\)</span> to obtain <span class="math inline">\(u\)</span></p>
</div>
<div id="tabset-3-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-2-tab">
<ol type="1">
<li>Variance correspondence</li>
</ol>
<p>Takin into account that</p>
<p><span class="math display">\[
\begin{align*}
\sigma^2_A &amp;= 2pq \, \sigma_a^2 + 2pq (q - p)^2 \, \sigma_d^2 \\
\sigma^2_{A^*} &amp;= 2pq \, \sigma_a^2 \\
\sigma^2_D &amp;= (2pq)^2 \, \sigma_d^2 \\
\sigma^2_{D^*} &amp;= 2pq (1 - 2pq) \, \sigma_d^2
\end{align*}
\]</span></p>
<p>We can then demonstrate that</p>
<p><span class="math display">\[
\begin{align*}
\sigma^2_A + \sigma^2_D &amp;= 2pq\sigma^2_a + 2pq(q-p)^2\sigma^2_d + (2pq)^2\sigma^2_d \\
&amp;= 2pq\sigma^2_a + \sigma^2_d(2pq(q-p)^2 + (2pq)^2)\\
&amp;= 2pq\sigma^2_a + \sigma^2_d(2pq(q^2 + q^2 -2pq) + 4p^2q^2) \\
&amp;= 2pq\sigma^2_a + \sigma^2_d(2pq^3 + 2p^3q -4p^2q^2 + 4p^2q^2) \\
&amp;= 2pq\sigma^2_a + \sigma^2_d(2pq(q^2 + p^3))\\
&amp;= 2pq\sigma^2_a + \sigma^2_d(2pq(1-2pq)) \\
&amp;= 2pq(\sigma^2_a + \sigma^2_d(1-2pq)
\end{align*}
\]</span></p>
<p><span class="math display">\[
\begin{align*}
\sigma^2_{A^*} + \sigma^2_{D^*} &amp;= 2pq \, \sigma_a^2 + 2pq (1 - 2pq) \, \sigma_d^2 \\
&amp;= 2pq(\sigma^2_a + \sigma^2_d(1-2pq)
\end{align*}
\]</span></p>
<p>And therefore</p>
<p><span class="math display">\[
\sigma^2_A + \sigma^2_D = \sigma^2_{A^*} + \sigma^2_{D^*} = 2pq(\sigma^2_a + \sigma^2_d(1-2pq)
\]</span></p>
<p>Which means that both parametrizations are equivalents</p>
</div>
</div>
</div>
</section>
</section>
<section id="autoteraploid-theory" class="level1">
<h1>Autoteraploid Theory</h1>
<p>The theoretical derivation that allows to partition variance into additive and dominance has been made under the breeding framework. No derivation has been published to partition genetic values into pure additive and pure dominance effect in polyploids.</p>
<p>The increased level of complexity because of the enhanced number of possible alleles is illustrated in the definition of the genotypic value. Not only the number of additive effects is increased but also the amount of interactions between alleles. Disregarding trigenic and quadrigenic effect, it can be defined as</p>
<p><span class="math display">\[
G_{ijkl} = \alpha_i + \alpha_j + \alpha_k + \alpha_l +  \beta_{ij} +
\beta_{ik} + \beta_{il} + \beta_{jk} + \beta_{jl} + \beta_{kl}
\]</span></p>
<p>Again, the residuals from the regression equation for additive effects <span class="math inline">\(y_{ijkl} = G_{ijkl} - (\mu + \alpha_i + \alpha_j + \alpha_k + \alpha_l)\)</span> is what we call the dominance deviations.</p>
<p>As stated in <a href="https://doi.org/10.1534/genetics.118.300685">Endelman et al, 2018</a>, the breeding value (BV) of an individual is defined as twice the mean genotypic value of its progeny relative to the population mean. Under the model assumptions, all six possible gene pairs for tetraploid genotype ijkl have equal frequency in its gametes which leads to the following expression</p>
<p><span class="math display">\[
BV_{ijkl} = 2[\frac{1}{6}(G_{ij..} + G_{ik..} + G_{il..} + G_{jk..} + G_{jl..} + G_{kl..}) - \mu]
\]</span></p>
<p>Taking into account that, as previously explained</p>
<p><span class="math display">\[
\beta_{ij} = G_{ij} - \mu - alpha_i - alpha_j
\]</span></p>
<p>We can solve for <span class="math inline">\(G_{ij} = \beta_{ij} + \alpha_i + \alpha_j + \mu\)</span>. Therefore</p>
<p><span class="math display">\[
\begin{align*}
BV_{ijkl}
&amp;= 2\left[\frac{1}{6}\Big(
(\beta_{ij} + \alpha_i + \alpha_j + \mu) +
(\beta_{ik} + \alpha_i + \alpha_k + \mu) +
(\beta_{il} + \alpha_i + \alpha_l + \mu) \right.\\
&amp;\qquad\qquad\left.+
(\beta_{jk} + \alpha_j + \alpha_k + \mu) +
(\beta_{jl} + \alpha_j + \alpha_l + \mu) +
(\beta_{kl} + \alpha_k + \alpha_l + \mu)
\Big) - \mu \right] \\[6pt]
&amp;= 2\left[\frac{1}{6}\Big(
6\mu + 3(\alpha_i + \alpha_j + \alpha_k + \alpha_l)
+ \beta_{ij} + \beta_{ik} + \beta_{il}
+ \beta_{jk} + \beta_{jl} + \beta_{kl}
\Big) - \mu \right] \\[6pt]
&amp;= (\alpha_i + \alpha_j + \alpha_k + \alpha_l)
+ \frac{1}{3}\left(
\beta_{ij} + \beta_{ik} + \beta_{il}
+ \beta_{jk} + \beta_{jl} + \beta_{kl}
\right) \\[6pt]
&amp;= u + \frac{1}{3}v .
\end{align*}
\]</span></p>
<p>Therefore, the breeding value equals the total additive value <span class="math inline">\(u\)</span> plus 1/3 of the total digenic dominance <span class="math inline">\(v\)</span>.</p>
<p>Based on a series of equations developed in the cited article, the equation <span class="math inline">\(0 = \sum_k p_k\beta_{ik}\)</span> yield the following two equalities</p>
<p><span class="math display">\[
p\beta_{BB} + q\beta_{Bb} = 0
\]</span></p>
<p><span class="math display">\[
p\beta_{Bb} + q\beta_{bb} = 0
\]</span></p>
<p>To solve this system of equation, the parameter <span class="math inline">\(\beta = \beta_{Bb} - \frac{1}{2}(\beta_{bb} + \beta_{BB})\)</span> is introduced, producing the following set of equalities</p>
<p><span class="math display">\[
\beta_{bb} = -2p^2\beta
\]</span></p>
<p><span class="math display">\[
\beta_{Bb} = 2pq\beta
\]</span></p>
<p><span class="math display">\[
\beta_{BB} = -2q^2\beta
\]</span></p>
<p>Thus, if we calculate the dominance deviation for the five possible genotypes of an autoetraploid (aaaa, Aaaa, AAaa, AAAa, AAAA) by examining all digenic interactions and we substitute based on the previous equations</p>
<p><span class="math display">\[
v =
\begin{cases}
v_{aaaa} = 6\beta_{bb} = -12p^2\beta \\[6pt]
v_{Aaaa} = 3\beta_{bb} + 3\beta_{Bb} = -6p^2\beta + 6pq\beta \\[6pt]
v_{AAaa} = \beta_{bb} + 4\beta_{Bb} + \beta_{BB} = -2p^2\beta + 8pq\beta - 2q^2\beta \\[6pt]
v_{AAAa} = 3\beta_{BB} + 3\beta_{Bb} = -6q^2\beta + 6pq\beta \\[6pt]
v_{AAAA} = 6\beta_{BB} = -12q^2\beta
\end{cases}
\]</span></p>
<p>If we substitute <span class="math inline">\(q\)</span> by <span class="math inline">\((p - 1)\)</span> we will obtain the following simplification</p>
<p><span class="math display">\[
v =
\begin{cases}
v_{aaaa} = \beta(-12p^2)\\
v_{Aaaa} = \beta(-12p^2 + 6p)\\
v_{AAaa} = \beta(-12p^2 + 12p -2)\\
v_{AAAa} = \beta(-12p^2 + 18p -6)\\
v_{AAAA} = \beta(-12p^2 + 24p - 12)\\
\end{cases}
\]</span></p>
<p>This expression can be generalized to any ploidy and dosage such that</p>
<p><span class="math display">\[
v = [-2\binom{\phi}{2}p^2 + 2p(\phi - 1)X -X(X-1)]\beta = Q\beta
\]</span></p>
<p>Notice that, for a diploid specie, following the recent derivation</p>
<p><span class="math display">\[
v_{\phi = 2} =
\begin{cases}
v_{aa} = -2p^2\\
v_{Aa} = 2pq)\\
v_{aa} = -2q^2
\end{cases}
\]</span></p>
<p>The digenic dominance matrix Dij is defined similarly as the covariance between dominance values relative to the dominance genetic variance, based on the expectation with respect to <span class="math inline">\(b \sim N(0, \sigma^2_{\beta})\)</span></p>
<p><span class="math display">\[
D_{ij} = \sigma_D^{-2}cov[v_i, v_j] = \sigma_D^{-2}cov[Q_i\beta, Q_j\beta] = \sigma_D^{-2}\sigma_{\beta}^{-2}Q_iQ_j
\]</span> <span class="math display">\[
\sigma^2_D = E[v^2] - E[v]^2 = E[v^2] = \binom{\phi}{2} 4\, p^2 q^2 \, \beta^2
\]</span></p>
<p>Therefore, the correct scaling of the Digenic Dominanc GRM is the following one</p>
<p><span class="math display">\[
\mathbf{D} = \frac{\mathbf{QQ^T}}{\binom{\phi}{2} \sum_k 4\, p_k^2 q_k^2}
\]</span></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("google");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../pages/03.1_predict_LMM.html" class="pagination-link" aria-label="Predictions in LMM">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Predictions in LMM</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../about.html" class="pagination-link" aria-label="About">
        <span class="nav-page-text">About</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© UW-Madison, 2025</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>This page is built with ❤️ and <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>